### PR TITLE
Adding support for 16UC1 image

### PR DESCRIPTION
**Pull request by lgeo3**
It is usefull to calibrate xtion infrared channel on Pepper robot

### CODE REVIEW

File: camera_calibration/src/camera_calibration/calibrator.py [hunk 1]
```diff
@@ -237,6 +237,10 @@ def mkgray(self, msg):
         Convert a message into a 8-bit 1 channel monochrome OpenCV image
         """
         # as cv_bridge automatically scales, we need to remove that behavior
+        if msg.encoding=='16UC1':
```
Comment (vrabaud) [inline camera_calibration/src/camera_calibration/calibrator.py]: please merge that condition with the one below to not have code duplication

**Comment by laurent-george** [comment]
depth_ir

File: camera_calibration/src/camera_calibration/calibrator.py [hunk 2]
```diff
@@ -237,9 +237,13 @@ def mkgray(self, msg):
...
         """
         # as cv_bridge automatically scales, we need to remove that behavior
-        if msg.encoding.endswith('16'):
-            mono16 = self.br.imgmsg_to_cv2(msg, "mono16")
-            mono8 = mono16.astype(numpy.uint8)
+        if msg.encoding == 'int16':
+            int16_image = self.br.imgmsg_to_cv2(msg)
+            mono8 = numpy.array(128 + numpy.clip(int16_image, 0, 127), dtype=numpy.uint8)
+            return mono8
+        if msg.encoding in ['16UC1', 'mono16', '16UC16']:
```
Comment (vrabaud) [inline camera_calibration/src/camera_calibration/calibrator.py:244]: 16UC16 is not a valid format

**Comment by vrabaud** [comment]
ping

**Pull request closed without merging**